llvm/include/llvm/Frontend/HLSL/HLSLRootSignature.h

```diff
@@ -198,6 +199,61 @@ class MetadataBuilder {
   SmallVector<Metadata *> GeneratedMetadata;
 };
 
+// RangeInfo holds the information to correctly construct a ResourceRange
+// and retains this information to be used for displaying a better diagnostic
+struct RangeInfo {
+  const static uint32_t Unbounded = static_cast<uint32_t>(-1);
```

Comment:
I think `~0U` (if you want to be concise) or `std::numeric_limits<uint32_t>::max()` are very slightly clearer here.